spec/acceptance/basic_webhook_spec.rb

```diff
@@ -30,38 +28,32 @@ class {'r10k::webhook':
       it { is_expected.to be_running }
     end
-
-    # rubocop:disable RSpec/RepeatedExample
-    shell('/usr/bin/curl -d \'{ "repository": { "name": "puppetlabs-stdlib" } }\' -H "Accept: application/json" "http://localhost:8088/module" -k -q') do |r|
-      it { expect(r.stdout).to match(%r{^.*success.*$}) }
-      it { expect(r.exit_code).to eq(0) }
+    describe command('/usr/bin/curl -d \'{ "repository": { "name": "puppetlabs-stdlib" } }\' -H "Accept: application/json" "http://localhost:8088/module" -k -q') do
+      its(:stdout) { is_expected.not_to match %r{.*You shall not pass.*} }
+      its(:exit_status) { is_expected.to eq 0 }
     end
-    shell('/usr/bin/curl -X POST -d \'{ "repository": { "full_name": "puppetlabs/puppetlabs-stdlib", "name": "PuppetLabs : StdLib" } }\' "http://localhost:8088/module" -k -q') do |r|
-      it { expect(r.stdout).to match(%r{^.*success.*$}) }
-      it { expect(r.exit_code).to eq(0) }
+    describe command('/usr/bin/curl -X POST -d \'{ "repository": { "full_name": "puppetlabs/puppetlabs-stdlib", "name": "PuppetLabs : StdLib" } }\' "http://localhost:8088/module" -k -q') do
+      its(:stdout) { is_expected.not_to match %r{.*You shall not pass.*} }
+      its(:exit_status) { is_expected.to eq 0 }
     end
-    shell('/usr/bin/curl -d \'{ "ref": "refs/heads/production" }\' -H "Accept: application/json" "http://localhost:8088/payload" -k -q') do |r|
-      it { expect(r.stdout).to match(%r{^.*success.*$}) }
-      it { expect(r.exit_code).to eq(0) }
+    describe command('/usr/bin/curl -d \'{ "ref": "refs/heads/production" }\' -H "Accept: application/json" "http://localhost:8088/payload" -k -q') do
+      its(:stdout) { is_expected.not_to match %r{.*You shall not pass.*} }
+      its(:exit_status) { is_expected.to eq 0 }
     end
-    shell('/usr/bin/curl -X POST -d \'%7b%22ref%22%3a%22master%22%7d\' "http://localhost:8088/payload" -q') do |r|
-      it { expect(r.stdout).to match(%r{^.*success.*$}) }
-      it { expect(r.exit_code).to eq(0) }
+    describe command('/usr/bin/curl -X POST -d \'%7b%22ref%22%3a%22maste%r22%7d\' "http://localhost:8088/payload" -q') do
+      its(:stdout) { is_expected.not_to match %r{.*You shall not pass.*} }
+      its(:exit_status) { is_expected.to eq 0 }
     end
-    shell('/usr/bin/curl -X POST -d \'{ "push": { "changes": [ { "new": { "name": "production" } } ] } }\' "http://localhost:8088/payload" -q') do |r|
-      it { expect(r.stdout).to match(%r{^.*success.*$}) }
-      it { expect(r.exit_code).to eq(0) }
+    describe command('/usr/bin/curl -X POST -d \'{ "push": { "changes": [ { "new": { "name": "production" } } ] } }\' "http://localhost:8088/payload" -q') do
+      its(:stdout) { is_expected.not_to match %r{.*You shall not pass.*} }
+      its(:exit_status) { is_expected.to eq 0 }
     end
-    # rubocop:enable RSpec/RepeatedExample
 
-    # rubocop:disable RSpec/MultipleExpectations
-    it 'successfully locks when hammered with multiple requests' do
```

Comment:
I do wonder about the concurrency here. describe blocks are run separate so you might be changing the test to run 4 commands sequentially while the test is actually run 4 commands in parallel.